File: partner-service/src/main/java/com/orbitbank/partnerservice/model/Partner.java
package com.orbitbank.partnerservice.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Document(collection = "partner")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties
public class Partner {
    @Id
    @Indexed(unique = true)
    private String partnerId;
    private String partnerUsername;
    private String partnerPassword;
    private String partnerFirstName;
    private String partnerLastName;
    private String partnerEmail;
    private String partnerInsuranceCompany;
    private String partnerSellerLicenceNo;
    private String partnerLicenseExpiry;
    private String partnerDOB;
    private String partnerPhone;
    private String partnerCity;
    private String partnerDistrict;
    private String partnerState;
    private String partnerCountry;
    private String partnerPinCode;
    private Boolean autorized=false;
}

File: bank-service/src/main/kotlin/com/orbitbank/bankservice/model/Partner.kt
package com.orbitbank.bankservice.model

import lombok.AllArgsConstructor
import lombok.Data
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.util.Date
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Data
@NoArgsConstructor
@AllArgsConstructor
class Partner (
    @Id
    @GeneratedValue
    var id: Int,
    var partnerInsuranceCompany: String,
    var partnerName: String,
    var partnerEmail: String,
    var partnerPassword: String,
    var partnerCertification: String,
    val partnerLicenseExpiry: Date,
    var partnerPolicyCount: Int
)
